### PR TITLE
ref: `ensure_integration_enabled` without original function

### PR DESCRIPTION
**Pull request by szokeasaurusrex**
`ensure_integration_enabled` and `ensure_integration_enabled_async` can now decorate functions that return `None` without an original function.

### CODE REVIEW

File: sentry_sdk/utils.py [hunk 1]
```diff
@@ -1657,25 +1689,49 @@ def patch_my_function():
...
+
+        return wraps(original_function)(runner)
 
     return patcher
 
 
+if TYPE_CHECKING:
+
+    @overload
+    def ensure_integration_enabled_async(
```
Comment (szokeasaurusrex) [inline sentry_sdk/utils.py:1717]: I am confused about why `mypy` is complaining about this overload, but not the ones for `ensure_integration_enabled`. Do you have any idea? Maybe I am missing something.

File: sentry_sdk/utils.py [hunk 2]
```diff
@@ -1657,25 +1689,49 @@ def patch_my_function():
             return my_function()
     ```
     """
+    if TYPE_CHECKING:
+        # Type hint to ensure the default funciton has the write typing. The overloads
```
Comment (sentrivana) [inline sentry_sdk/utils.py:1693]:
```suggestion
        # Type hint to ensure the default function has the right typing. The overloads
```

File: sentry_sdk/utils.py [hunk 3]
```diff
@@ -1684,17 +1740,24 @@ def ensure_integration_enabled_async(
     Please refer to the `ensure_integration_enabled` documentation for more information.
     """
 
+    if TYPE_CHECKING:
+        # Type hint to ensure the default funciton has the write typing. The overloads
```
Comment (sentrivana) [inline sentry_sdk/utils.py:1744]:
```suggestion
        # Type hint to ensure the default function has the right typing. The overloads
```

**Review by sentrivana — APPROVED** [review]
The more I see of this advanced typing business the less I understand it 😆 But in addition to being the decorator master you're now also the typing master, so I trust this works and gives nice type hints as intended.

Found a couple typos, please fix, otherwise lgtm.

File: sentry_sdk/utils.py (hunk 1 above)
Comment (sentrivana) [reply sentry_sdk/utils.py:1717]: Sorry, no idea. Agree that it looks exactly like the sync case. Maybe we can just ignore the error?

File: sentry_sdk/utils.py (hunk 2 above)
Comment (szokeasaurusrex) [reply sentry_sdk/utils.py:1693]: lol

**Comment by szokeasaurusrex** [comment]
> The more I see of this advanced typing business the less I understand it 😆 But in addition to being the decorator master you're now also the typing master, so I trust this works and gives nice type hints as intended.

Yeah, at first glance it appears to work. Of course, if you or anyone else notices weird hinting behavior, let me know, and I can look into fixing it.

**Pull request merged**